lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -367,7 +368,12 @@ def add_stored_credential(xml, options)
 
       def merchant_initiated_unscheduled(options)
         return options[:merchant_initiated_unscheduled] if options[:merchant_initiated_unscheduled]
-        return 'Y' if options.dig(:stored_credential, :initiator) == 'merchant' && options.dig(:stored_credential, :reason_type) == 'unscheduled'
+        return 'Y' if options.dig(:stored_credential, :initiator) == 'merchant' && options.dig(:stored_credential, :reason_type) == 'unscheduled' || options.dig(:stored_credential, :reason_type) == 'recurring'
+      end
+
+      def recurring_transaction(options)
+        return options[:ssl_add_token] if options.dig(:stored_credential, :reason_type) == 'recurring' && options.dig(:store_credential, :initial_transaction) == true
+        return 'Y'
```

Comment:
So, if the condition above isn't met, this method will always return 'Y'. Is that what we want?

Reply:
There must be something amiss in the code because as it is a 'Y' is sent no matter what as long as it is a recurring purchase.  This doesn't present an issue because if it is sent on a subsequent call it just registers it as an update without changing the token. But I am tinkering with it now.

Reply:
Conditional added plus a new test and some clean up!